rwengine/include/script/ScriptTypes.hpp

```diff
@@ -66,13 +68,20 @@ struct SCMOpcodeParameter {
 
 	int integerValue() const 
 	{
-		if ( type == TGlobal )
-		{
-			return * globalInteger;
-		}
-		else
+		switch (type)
 		{
-			return integer;
+		case TGlobal:
+		case TLocal:
+			return *globalInteger;
```

Comment:
Yes. This still isn't quite right. I'll fix it further up in the VM so it doesn't need casting here.